runtime/src/precompiles.rs

```diff
@@ -2,13 +2,58 @@ use pallet_evm::{Precompile, PrecompileHandle, PrecompileResult, PrecompileSet};
 use sp_core::H160;
 use sp_std::marker::PhantomData;
 
+use pallet_evm_precompile_blake2::Blake2F;
+use pallet_evm_precompile_bn128::{Bn128Add, Bn128Mul, Bn128Pairing};
+use pallet_evm_precompile_ed25519::Ed25519Verify;
 use pallet_evm_precompile_modexp::Modexp;
 use pallet_evm_precompile_sha3fips::Sha3FIPS256;
 use pallet_evm_precompile_simple::{ECRecover, ECRecoverPublicKey, Identity, Ripemd160, Sha256};
+use pallet_evm_precompile_sr25519::Sr25519Precompile;
+use pallet_evm_precompile_substrate_ecdsa::SubstrateEcdsaPrecompile;
 use pallet_evm_precompile_xvm::XvmPrecompile;
 
 pub struct FrontierPrecompiles<R>(PhantomData<R>);
 
+pub mod consts {
+	use sp_core::H160;
+
+	pub const EC_RECOVER: H160 = hash(1);
+	pub const SHA256: H160 = hash(2);
+	pub const RIPEMD160: H160 = hash(3);
+	pub const IDENTITY: H160 = hash(4);
+	pub const MODEXP: H160 = hash(5);
+	pub const BN128_ADD: H160 = hash(6);
+	pub const BN128_MUL: H160 = hash(7);
+	pub const BN128_PAIRING: H160 = hash(8);
+	pub const BLAKE2F: H160 = hash(9);
+	pub const SHA3_FIPS256: H160 = hash(0x400);
+	/// 402 is used in Astar and Moonbeam, so preserve the address for contracts interoperability
+	pub const EC_RECOVER_PUBLIC_KEY: H160 = hash(0x402);
+	/// 403 is also used in Astar, so preserve the address for contracts interoperability.
+	/// TO DECIDE: Celo uses 0xF3 for it, so we need to decide which one to use.
+	pub const ED25519_VERIFY: H160 = hash(0x403);
+	/// 5002 is used in Astar, so preserve the address for contracts interoperability
+	pub const SR25519_VERIFY: H160 = hash(0x5002);
+	/// 5003 is used in Astar, so preserve the address for contracts interoperability
+	pub const ECDSA_VERIFY: H160 = hash(0x5003);
+	/// 5005 is used in Astar, so preserve the address for contracts interoperability
+	pub const XVM: H160 = hash(0x5005);
+
+	const fn hash(a: u64) -> H160 {
+		let bytes = a.to_be_bytes();
+		let mut result = [0u8; 20];
+		result[12] = bytes[0];
+		result[13] = bytes[1];
+		result[14] = bytes[2];
+		result[15] = bytes[3];
+		result[16] = bytes[4];
+		result[17] = bytes[5];
+		result[18] = bytes[6];
+		result[19] = bytes[7];
+		H160(result)
+	}
```

Comment:
We should have some unit tests to verify this is the same as  `H160::from_low_u64_be`.

Consider using a [proptest][1] like this:

```rust
#[proptest]
fn hash_function_is_correct(a: u64) {
    assert_eq!(hash(a), H160::from_low_u64_be(a));
}
```

[1]: https://docs.rs/test-strategy/latest/test_strategy/#example